### PR TITLE
Download API enabled. logbin variables renamed. Pileup of CTCF around boundaries. 

### PR DESCRIPTION
**Pull request by agalitsyna**
1. Download with cooltools API added for each notebook. 
2. contacts_vs_distance.ipynb: logbin expected names are renamed consistent with recent change in cooltools: x->diag.avg and Pc->balanced.avg
3. Insulation_and_boundaries.ipynb: pileup of CTCF around boundaries added.

### CODE REVIEW

File: Insulation_and_boundaries.ipynb [hunk 1]
```diff
@@ -44,6 +44,27 @@
     "from cooltools.insulation import calculate_insulation_score, find_boundaries"
```
Comment (Phlya) [inline Insulation_and_boundaries.ipynb:44]: <p>What do others think, should we call average ChIP-seq profile a pileup, or is it confusing, when we use the same word for Hi-C?</p><p> </p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/open2c/open2c_examples/pull/5/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='21'/>

**Comment by Phlya** [comment]
Looks great, and a good start for combining insulation with CTCF ChIP-seq!

File: Insulation_and_boundaries.ipynb (hunk 1 above)
Comment (gfudenberg) [reply Insulation_and_boundaries.ipynb:44]: <p>I think using the same word is good, but maybe we could clarify as 1D pileup or 2D pileup</p>

Comment (golobor) [reply Insulation_and_boundaries.ipynb:44]: +1  , i'm okay with 1D pileups vs 2d pileups - they have exactly same math, right?

Comment (agalitsyna) [reply Insulation_and_boundaries.ipynb:44]: <p>Clarified it in the annotation, ok!</p>

Comment (mimakaev) [reply Insulation_and_boundaries.ipynb:44]: yeah, it's confusing - all this pileup - stackup - etc terminology. Do people outside of our circle even use terms "pileup" and/or "stackup"?

Comment (Phlya) [reply Insulation_and_boundaries.ipynb:44]: <p>No. My examiner (computational biologist!) was confused by "pileup". People only know "pileup" for pileup of reads: http://www.htslib.org/doc/samtools-mpileup.html</p><p> </p>

File: pileup_CTCF.ipynb [hunk 2]
```diff
@@ -0,0 +1,649 @@
+{
```
Comment (Phlya) [inline pileup_CTCF.ipynb:1]: <p>I'd say, for species with telocentric chromosomes (e.g. mouse)...</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/open2c/open2c_examples/pull/5/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='8'/>

Comment (Phlya) [inline pileup_CTCF.ipynb:1]: <p>Ha I am not used at looking at just observed pileups. Maybe vmin should be higher?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/open2c/open2c_examples/pull/5/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='20'/>

**Comment by gfudenberg** [comment]
I think a more commonly used name for averages is a meta prefix, e.g.
metagene plot. But I haven't seen it used for HiC

File: pileup_CTCF.ipynb (hunk 2 above)
Comment (agalitsyna) [reply pileup_CTCF.ipynb:1]: Yeah, I also do not use it. Yet it seems to be important for the demonstration of functionality. How about now?

Comment (agalitsyna) [reply pileup_CTCF.ipynb:1]: Ah, sure! Good point!

**Pull request merged**